addons/website/static/src/js/widgets/theme.js

```diff
@@ -699,6 +719,33 @@ var ThemeCustomizeDialog = Dialog.extend({
             $dropdown.find('.o_theme_customize_dropdown_btn').remove();
             $dropdown.prepend($btn);
         });
+        _.each(this.$('.o_theme_customize_auto'), function (el) {
+            var $el = $(el);
+            var $customizeOption = $el.closest('.o_theme_customize_option');
+            var values = [
+                {'variable': 'headings-font-number', 'property': '--font-number'},
+                {'variable': 'font-number', 'property': '--headings-font-number'},
+                {'variable': 'buttons-font-number', 'property': '--buttons-font-number'},
+                {'variable': 'navbar-font-number', 'property': '--navbar-font-number'},
+            ];
+            $customizeOption.removeClass('active');
+            _.each(values, function (value) {
+                if ($el.data('variable') === value.variable) {
+                    if ($el.data('value') === parseInt(self.style.getPropertyValue(value.property))) {
+                        $customizeOption.addClass('active');
+                        $el.closest('.o_theme_customize_selection').siblings('.o_theme_customize_dropdown_btn').remove();
+                        var classes = 'btn btn-light dropdown-toggle w-100 o_theme_customize_dropdown_btn o_theme_customize_option_font_' + $el.data('value');
+                        var $btn = $('<button/>', {
+                            type: 'button',
+                            class: classes,
+                            'data-toggle': 'dropdown',
+                        });
+                        $el.closest('div.o_theme_customize_dropdown').prepend($btn);
+                    }
+                }
+            });
```

Comment:
Come on Andrea... you are wasting time writing so much code you know yourself is wrong :/ Ask me if I am not clear enough.

Here your code seems to say "for each auto option, define a set of strange variable/property pairs then for each one, check if it is the one of the current option and ..."... why :/

You just had to say "for each option, check if the related variable value has the value the option would set". It is basically just removing lines of what you did, why would you need to know about the names of the variables in this method ?
For each option something like `$el.data('value') === self.style.getPropertyValue('--' + $el.data('variable'))` seems enough to tell you if you need to toggle the active class or not... this is one line :/

Reply:
Yes indeed this was totally useless.. This is sad because at some point I ended with this same line but changed for other useless stuff.. My bad, plus I wasn't in very great shape this afternoon. Anyway, this is much better like that ! This said, I still need to rebuild my button with the right font in it instead of letting "Bootstrap" everywhere when it's not the case.